ENIGMAsystem/SHELL/Platforms/SDL/Joystick.cpp

```diff
@@ -92,15 +91,35 @@ void joystick_uninit() {
 }
   
 void joystick_update() {     
-  if (enigma_user::joystick_exists(1) || enigma_user::joystick_exists(2))
-    update = true;
-  else if (update) {
-    joystick_uninit();
-    if (joystick_init())
-      update = false;
+  if (SDL_NumJoysticks() == 0) {
+    if (joystick1 != NULL) {
+      SDL_JoystickClose(joystick1);
+      joystick1 = NULL;
+    }
+
+    if (joystick2 != NULL) {
+      SDL_JoystickClose(joystick2);
+      joystick2 = NULL;
+    }
   }
-  if (update)
-    SDL_JoystickUpdate();
+
+  if (SDL_NumJoysticks() == 1) {
+    if (joystick1 == NULL)
```

Comment:
This logic doesn't account for joystick 1 being unplugged. Does SDL handle this?